sdv/single_table/base.py

```diff
@@ -610,6 +611,122 @@ class BaseSingleTableSynthesizer(BaseSynthesizer):
     for all single-table synthesizers.
     """
 
+    def __init__(
+        self,
+        metadata,
+        enforce_min_max_values=True,
+        enforce_rounding=True,
+        locales=['en_US'],
+    ):
+        super().__init__(metadata, enforce_min_max_values, enforce_rounding, locales)
+        self._invalid_patterns = []
+        self._valid_patterns = []
+
+    def add_cag(self, patterns):
+        """Add the list of constraint-augmented generation patterns to the synthesizer.
+
+        Args:
+            patterns (list):
+                A list of CAG patterns to apply to the synthesizer.
+        """
+        if not hasattr(self, '_original_metadata'):
+            self._original_metadata = self.metadata
+
+        for pattern in patterns:
+            self.metadata = pattern.get_updated_metadata(self.metadata)
```

Comment:
Will this work if we have a pattern that we have to have fallback to reject sampling?